releasenotes/notes/improve-mappers-b55cb0ca5fd656e4.yaml

```diff
@@ -0,0 +1,21 @@
+---
+features:
+  - |
+    The class :class:`.second_q.mappers.ModeBasedMapper` has been added to implement mode based
+    mapping via a Pauli table (previously part of :class:`.second_q.mappers.QubitMapper`).
+upgrade:
```

Comment:
```suggestion
```
`upgrade` is supposed to be things an end user needs to do to upgrade (alter) their code for the release e.g. things have been removed and users code needs to be adjusted. I would just remove this and make it all part of the features statement - though honestly this is supposed to be more an end user facing statement around the changes where I am not sure how much, if anything, this internal change affects them or surfaces to them in any way. I imagine code they had before using the mappers works just as it did before.

Reply:
That explanation makes sense. I have removed the point about changes in caching, and the inheritance which I believe has no end user facing consequences. The removal of public methods as pointed out by @mrossinek should obviously stay. Based on your reasoning, the info that `pauli_table` is now an instance method instead of a class method should stay, I guess, in case someone is calling it on a class in their code.